src/Helper/GridHelperService.php

```diff
@@ -336,9 +336,9 @@ public function getFilterCondition(string $filterJson, ClassDefinition $class, ?
                     } elseif (in_array($filterField, $systemFields)) {
                         // system field
                         if ($filterField == 'fullpath') {
-                            $conditionPartsFilters[] = 'concat(`path`, `key`) ' . $operator . ' ' . $db->quote('%' . $filter['value'] . '%');
+                            $conditionPartsFilters[] = 'concat(lower(`path`), lower(`key`)) ' . $operator . ' lower(' . $db->quote('%' . $filter['value'] . '%') . ')';
```

Comment:
```suggestion
                           // add the following variable to line 337 or outer part of the IF block so that it could be used in multiple places eg. for `fullpath` and `key`
                            $lowerCasedFilterValue = strtolower($filter['value']);
                            
                            
                            // this should be slightly more efficient, as it lowercasing once for the concat and no need to lowecase the $filter['value'] multiple times in the DB.
                            $conditionPartsFilters[] = 'lower(concat(`path`, `key`)) ' . $operator .' ' . $db->quote('%' . lowerCasedFilterValue . '%');
```

However, i have some concern that we are only applying this changes to `key` and `fullpath`, it would end up to be inconsistent. 

The best would be instead something general (covering all the fields) and "toggle-able", eg. with a checkbox could decide wheter should be case sensitive or not.

Also the issue was also about the wildcard `*`, so we might need to replace `*` with `%` in `$filter['value']` eg. searching for "AU*IA" would result in `LIKE "%AU%IA%"` (when searching for Austria and Australia 😄 )

Reply:
Thank you for the feedback! I pushed an change with:
- The lowercasing for the value is done once in PHP now instead of with SQL `lower()`
- I have added wildcard support

> However, i have some concern that we are only applying these changes to key and fullpath, it would end up to be inconsistent.

I looked into this and it seems a change should be made in the following file in the main Pimcore repository:
https://github.com/pimcore/pimcore/blob/11.x/models/DataObject/ClassDefinition/Data.php:442
```
        if (!$isNumeric) {
            if ($value === 'NULL') {
                if ($operator === '=') {
                    $operator = 'IS';
                } elseif ($operator === '!=') {
                    $operator = 'IS NOT';
                }
            } elseif (!is_array($value) && !is_object($value)) {
                $lowerCasedValue = strtolower($value);
                if ($operator === 'LIKE') {
                    $key = strtolower($key); // case insensitive search
                    $value = str_replace('*', '%', $lowerCasedValue); // wildcard support
                    $value = $db->quote('%' . $value . '%');
                } else {
                    $value = $db->quote($lowerCasedValue);
                }
            }
        }
```
This would add case insensitivity + support for wildcards for the regular text-type fields. If this looks OK to you I can make a PR there also.

> The best would be instead something general (covering all the fields) and "toggle-able", eg. with a checkbox near the search term box to decide whether it should be case sensitive or not.

I am willing to try this, but could not locate the correct files/templates to change the UI for this. Any pointers to the correct files or changes to be made would be welcome.

Reply:
@robin-ntriga  Thanks for implementing the feedback. 
We would greatly appreciate the PR in the core for all fields.

I think we can skip the toggle-able filter, because this filter is provided by ExtJS https://github.com/pimcore/admin-ui-classic-bundle/blob/e9ad0daae2e9a75f9190eae15e17b98758ee85a3/public/js/pimcore/object/helpers/grid.js#L275 and with your changes it's case insensitive everywhere.